text/0008-weighted-uniform-random-tip-selection/0008-weighted-uniform-random-tip-selection.md

```diff
@@ -0,0 +1,190 @@
++ Feature name: `weighted-uniform-random-tip-selection`
++ Start date: 2020-03-09
++ RFC PR: [iotaledger/protocol-rfcs#0008](https://github.com/iotaledger/protocol-rfcs/pull/0008)
+
+# Summary
+
+Weighted Uniform Random Tip Selection on a subset enables a node to perform fast tip-selection to increase transaction throughput.
+The algorithm selects tips which are non-lazy in order to maximize confirmation rate.
+
+# Motivation
+
+Because of the `white-flag` confirmation algorithm, it is no longer necessary to perform complex
+tip-selection which evaluates ledger mutations while walking, therefore, a simpler and better 
+performing algorithm can be used to select tips, which in turn increases overall transaction throughput.
+
+In order to maximize confirmation rate however, the algorithm needs to return tips which are `non-lazy`.
+Non-lazy in this context means, that a tip does not attach to a cone of transactions which is too far
+in the past, as such cone is likely to be already confirmed and therefore doesn't contribute to the
+rate of newly confirmed transactions when a milestone is issued.
+
+# Detailed design
+
+Definitions:
+* `Direct Approvers` are the set of transactions which directly approve a given transaction.
+* `Approvee` is the directly approved transaction of a given transaction.
+Example: the trunk/branch transactions are the approvees of a given transaction.
+* `Solid` means that the past cone of a given transaction exists in the database.
+* A `tail transaction` is the transaction at index zero of a bundle. Only a tail transaction can be a `tip`.
+* A `tip` is a solid tail transaction of a bundle without any approvers. Its past cone contains only structural valid bundles.
+* A `score` is a scoring  determining the likeliness to select a given `tip`.
+* `Confirmed Root Transactions` defines the set of first seen transactions which are confirmed by a previous milestone 

```

Comment:
It should be made clear that the milestone itself is also a "Confirmed Root Transactions" since it effectively confirms itself. Otherwise the linear tangle, i.e. branch always equal to trunk, would not have any Confirmed Root Transactions, rendering the tip score computation impossible.

Reply:
Added a note.